include/core/CFastMutex.h

```diff
@@ -76,7 +76,7 @@ class CORE_EXPORT CFastMutex : private CNonCopyable {
 #ifdef Windows
     SRWLOCK m_Mutex;
 #elif defined(MacOSX)
-    OSSpinLock m_Mutex;
+    os_unfair_lock m_Mutex;
```

Comment:
How does the performance compare?  The [docs](https://developer.apple.com/documentation/os/1646466-os_unfair_lock_lock) say:

> This is a replacement for the deprecated OSSpinLock. This function doesn't spin on contention, but instead waits in the kernel to be awoken by an unlock.

To me this sounds like there's a chance of no benefit over `pthread_mutex_t`, in which case the implementation could be shared with Linux.

There are tests in `CReadWriteLockTest` that test the performance of all the different types of lock, not just read/write.  If you dig out some old CI logs and look at the timestamps on `Starting fast mutex lock throughput test at` and `Finished fast mutex lock throughput test at` for a macOS build then you should be able to compare the performance of the old and new types.  You can also do this on your laptop.

Reply:
With respect to performance. I tested locally on my laptop (MacBook Pro 2017, 2.9GHz Quad-Core Intel Core i7) and increased the number of iterations from 1000000 to 100000000. The test `CReadWriteLockTest/testPerformanceVersusMutex` was then run in a loop 10 times and the reported run times averaged.

Timings for `CFastMutex`:
* using `OSSpinLock` - average time taken = 2.9s
* using `os_unfair_lock` - average time taken = 9.1s
* using `pthread_mutex_t` (same as Linux)- average time take = 20.6s

So while the move to `os_unfair_lock` is appreciably slower than the deprecated `OSSpinLock` it still out strips `pthread_mutex_t`.

Reply:
I also had a look at what happened on macOS on ARM.

Before (from the PR CI build of #2137):

```
2021-12-01 09:26:10,389468 UTC [94129] INFO  CReadWriteLockTest.cc@357 Starting fast mutex lock throughput test at 01:26:10
2021-12-01 09:26:10,462001 UTC [94129] INFO  CReadWriteLockTest.cc@373 Finished fast mutex lock throughput test at 01:26:10
```

462001 - 389468 = 72533

After (from this PR's CI):

```
2021-12-03 10:13:57,598510 UTC [27626] INFO  CReadWriteLockTest.cc@357 Starting fast mutex lock throughput test at 02:13:57
2021-12-03 10:13:57,684028 UTC [27626] INFO  CReadWriteLockTest.cc@373 Finished fast mutex lock throughput test at 02:13:57
```

684028 - 598510 = 85518

It's ideal because there are too few iterations to dominate OS process scheduling effects, but those numbers are very close, so maybe the new mutex is more efficient on ARM.  Since ARM is the future for macOS we might as well get rid of the deprecated function usage.

Reply:
Agreed. 

I've just noticed that I've omitted to update the comments in `CFastMutex.h`. I'll do so now.